WebSharper.Owin/Owin.Sitelets.fs

```diff
@@ -81,35 +126,50 @@ module private Internal =
             | "trace" -> Http.Method.Trace
             | s -> Http.Method.Custom s
 
-        let Headers (headers: IHeaderDictionary) : seq<Http.Header> =
+        let Headers (headers: HeaderDictionary) : seq<Http.Header> =
             seq {
                 for KeyValue(k, vs) in headers do
                     for v in vs do
                         yield Http.Header.Custom k v
             }
 
-        let Query (query: IReadableStringCollection) : Http.ParameterCollection =
-            Http.ParameterCollection(
-                seq {
-                    for KeyValue(k, vs) in query do
-                        for v in vs do
-                            yield (k, v)
-                }
-            )
-
-        let Cookies (cookies: RequestCookieCollection) : HttpCookieCollection =
+        let Query (query: Arachne.Uri.Query) : Http.ParameterCollection =
+            match fst Arachne.Uri.Query.Pairs_ query with

```

Comment:
Yes, I thought so, as well. I'm using the pre-release 2.3.27-b066 so that I can use the Cookie support.